src/provider/utils/SemVer.ts

```diff
@@ -0,0 +1,114 @@
+export enum SemVerType {
+    VALID = 'VALID',
+    INVALID = 'INVALID',
+    UNKNOWN = 'UNKNOWN'
+}
+
+export enum Operator {
+    LESS_THAN = '<',
+    GREATER_THAN = '>',
+    LESS_THAN_OR_EQUAL = '<=',
+    GREATER_THAN_OR_EQUAL = '>='
+}
+
+/**
+ * Util for comparing semantic version numbers.
+ *
+ * NOTE: Only parses the major, minor, patch and pre-release parts of the semver string - build metadata is ignored.
+ *
+ * Spec: https://semver.org/
+ */
+export class SemVer {
+    private static readonly REGEX_SEMVER: RegExp = /(\d+)\.(\d+)\.(\d+)(?:-([^-+]+))?/;
+    private static readonly REGEX_INTEGER: RegExp = /^\d+$/;
+    private static readonly CACHE: {[key: string]: SemVer} = {};
+
+    public static compare(a: string | SemVer, operator: Operator, b: string | SemVer): boolean {
+        return SemVer.parse(a).compare(operator, b);
+    }
+
+    public static parse(version: string | SemVer): SemVer {
+        if (typeof version === 'string' || version === null || version === undefined) {
+            let semver = SemVer.CACHE[version];
```

Comment:
This cache looks like a premature optimisation - do we really think that it's worth it?

Reply:
Could be... Was motivated by the size of the regex, and how we're basically only ever going to need to check a rather small set of semver strings over and over again.